source/ldap-setup.rst

```diff
@@ -0,0 +1,361 @@
+         
+         .. code-block:: yaml
+
+            security:
+              authorization: "enabled"
+              ldap:
+                servers: "ldap.example.com"
+                transportSecurity: tls
+                authz:
+                   queryTemplate: "dc=percona,dc=com??sub?(&(objectClass=groupOfNames)(member={PROVIDED_USER}))"
+                userToDNMapping: 
+                      [
+                        {
+                          match: "([^@]+)@percona\\.com",
+                          ldapQuery: "dc=percona,dc=com??sub?(&(objectClass=organizationalPerson)(cn={0}))" 
+                        }
+                      ]
+
+            setParameter:
+              authenticationMechanisms: "PLAIN"
+
+         The {USER} variable substitutes the username transformed during the userToDNMapping stage.
+         
+         Modify the given example configuration to match your deployment, For example, replace ``ldap.example.com`` with the hostname of your LDAP server. Replace the domain controllers (DC) ``percona`` and ``com`` with those relevant to your organization. Depending on your LDAP schema, further modifications of the LDAP query may be required. 
+
+      #. Restart the ``mongod`` service:
+         
+         .. code-block:: bash
+
+            $ sudo systemctl restart mongod
+
+      #. Test the access to |PSMDB|:
+         
+         .. code-block:: bash
+
+            mongo -u "alice" -p "secretpwd" --authenticationDatabase '$external' --authenticationMechanism 'PLAIN'
+
+.. _active-directory:
+
+Active Directory configuration
+------------------------------
+
+Microsoft Active Directory uses a different schema for user and group definition. To illustrate |PSMDB| configuration, we will use the following :abbr:`AD (Active Directory)` users:
+
+.. code-block:: text
+
+   dn:CN=alice,CN=Users,DC=testusers,DC=percona,DC=com
+   userPrincipalName: alice@testusers.percona.com
+   memberOf: CN=testusers,CN=Users,DC=percona,DC=com
+
+   dn:CN=bob,CN=Users,DC=otherusers,DC=percona,DC=com
+   userPrincipalName: bob@otherusers.percona.com
+   memberOf: CN=otherusers,CN=Users,DC=percona,DC=com
+
+The following are respective :abbr:`AD (Active Directory)` groups:
+
+.. code-block:: text
+
+   dn:CN=testusers,CN=Users,DC=percona,DC=com
+   member:CN=alice,CN=Users,DC=testusers,DC=example,DC=com
+
+   dn:CN=otherusers,CN=Users,DC=percona,DC=com
+   member:CN=bob,CN=Users,DC=otherusers,DC=example,DC=com
+
+Use one of the given |PSMDB| configurations for user authentication and authorization in Active Directory:
+
+.. tabs::
+
+   .. tab:: No username transformation
+
+      1. Edit the ``mongod`` configuration file:
+         
+         .. code-block:: yaml
+
+            ldap: 
+              servers: "ldap.example.com" 
+              authz: 
+                queryTemplate: "DC=percona,DC=com??sub?(&(objectClass=group)(member:1.2.840.113556.1.4.1941:={USER}))" 
```

Comment:
Should be PROVIDED_USER since there no username transformation

Reply:
Fixed